### ISSUE TITLE
[Fleet] Add tests for System Integration package (all of the various metrics and logs that can be collected)

### ISSUE THREAD

**Issue by EricDavisX** [elastic/e2e-testing#284]
This ticket is implemented to go deeper on what support is already in the Ingest Manager / Fleet & Agent tests.  We have little so far, and we need more.  So I'll itemize the items that I see in the Integration that we can add.  

Note, not all of the below are turned on by default, so a policy change in those cases is required as part of the scenario.  To be safer, we could put in a 'set this option if not set' step.  Asserting which are on or off by default itself could be a useful test, but maybe not very interesting or error prone (not sure).  

-------------
Logs:
[ ] System auth logs (log) - /var/log/auth.log*  or /var/log/secure*
[ ] System syslog logs (log) - /var/log/messages* or /var/log/syslog*

Metrics
[ ] System core metrics
[ ] System cpu metrics
[ ] System diskio metrics
[ ] System entropy metrics
[ ] System filesystem metrics
[ ] System fsstat metrics
[ ] System load metrics
[ ] System network metrics
[ ] System network summary metrics
[ ] System process_summary metrics
[ ] System raid metrics
[ ] System service metrics
[ ] System socket metrics
[ ] System socket_summary metrics
[ ] System uptime metrics
[ ] System users metrics

**Comment by EricDavisX** [comment]
@ycombinator FYI, I don't think you have System tested yet on your side? I am happy to reduce scope, etc if you do!  And also, any lessons learned you might pass on from prior Integration work?

**Comment by ycombinator** [comment]
@EricDavisX That's correct — @mtojek and I just created an MVP for the system test runner for packages: https://github.com/elastic/elastic-package/pull/64. The next step for us is to use this runner to system test the Apache package: https://github.com/elastic/integrations/pull/263. After that we're going to tackle the System package, which will require some enhancements to the system test runner, most notably https://github.com/elastic/elastic-package/issues/89. 

Given that we're also working on other projects at the moment (e.g. package migration assessments), I'd suggest going ahead with this issue here. Or if you want, you can check where we are right before starting work, in case we've managed to make decent progress and are pretty close to having system tests for the System package. In the worst case we'll have two independent test runners testing the System package end-to-end, which isn't the worst outcome for a package that's enabled by default. WDYT?

**Comment by EricDavisX** [comment]
That's a perfect plan, thanks for the reply and the links.  Timeframe for when we'd get more of those checked-off is genuinely TBD, things are moving fairly slowly from our side.  Keep in touch.

**Comment by muskangulati-qasource** [comment]
Hi @EricDavisX,

We validated this ticket on latest 7.10 SNAPSHOT.

**Build details** :
```
Elastic Cloud Staging : 7.10.0
Elastic-Agent URL : https://artifacts-api.elastic.co/v1/search/7.10.0-SNAPSHOT
- Windows : elastic-agent-7.10.0-SNAPSHOT-windows-x86_64.zip
- Linux : elastic-agent-7.10.0-SNAPSHOT-linux-x86_64.tar.gz
- Mac : elastic-agent-7.10.0-SNAPSHOT-darwin-x86_64.tar.gz
```
## Steps :
1. Downloaded the Elastic-agent for respective OS.
2. Deployed the agent on 7.10 cloud build.
3. Added the System Integration in the policy applied on the endpoint.
4. Checked for the system integrations showing up for each OS from the Data Stream tab.

## Observations:
1 Windows :
**Screenshot** :
![windows_systemintegration](https://user-images.githubusercontent.com/60252716/93211009-4132a700-f77e-11ea-80d7-5eb413eb2803.PNG)

**Logs** :
- [ ] System auth logs (log) - /var/log/auth.log*  or /var/log/secure*
- [ ] System syslog logs (log) - /var/log/messages* or /var/log/syslog*

**Metrics** :
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics

2 Linux :
**Screenshot** :
![Linux_debrian_systemintegration](https://user-images.githubusercontent.com/60252716/93211037-4859b500-f77e-11ea-9083-6fd58793589e.PNG)

**Logs** :
- [ ] System auth logs (log) - /var/log/auth.log*  or /var/log/secure*
[Linux_auth.zip](https://github.com/elastic/e2e-testing/files/5224847/Linux_auth.zip)

- [ ] System syslog logs (log) - /var/log/messages* or /var/log/syslog*
[Linux_messages.zip](https://github.com/elastic/e2e-testing/files/5224848/Linux_messages.zip)

**Metrics** :
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics


3 Mac :
**Screenshot** :
![Mac_catalina_systemintegration](https://user-images.githubusercontent.com/60252716/93211043-4abc0f00-f77e-11ea-8c5e-8effce066dbb.PNG)

**Logs** :
- [ ] System auth logs (log) - /var/log/auth.log*  or /var/log/secure*
- [ ] System syslog logs (log) - /var/log/messages* or /var/log/syslog*

**Metrics** :
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics

Testcase(s) created :
- Windows : https://elastic.testrail.io/index.php?/cases/view/28380
- Linux : https://elastic.testrail.io/index.php?/cases/view/28382
- Mac : https://elastic.testrail.io/index.php?/cases/view/28383

Please let us know if we are missing anything.

Thanks!

**Comment by EricDavisX** [comment]
@muskangulati-qasource hi - thanks so much, this is a wonderful start to see.  I would like to dive even deeper, let me elaborate more testing details and we can keep processing on as time allows:

For each Agent OS of Windows, Linux and macOS let us turn on all of the supported datasets and test (what is tested above I think is just the defaults).  The list of supported data for each OS is part of the Ingest Manager 'summary' for the Integration, for example, here: 
https://kibana.endpoint.elastic.dev/app/ingestManager#/integrations/detail/system-0.5.3/overview
 - I see listed for each Dataset type what OSes it is supported on.  We need only test one variant of each of the 3 main OS (win, linux mac) we can skip FreeBSD, for example.

I would suggest that we run the test individually for each OS (with only 1 Agent installed, the OS being tested - to ensure evaluation of just the data from that Agent)

Then after seeing all dataset types, we can turn all off but one (randomly I will say we can pick System 'CPU' Metrics, which is supported on al 3 OSes.  All but the one datastream should stop reporting new data.  If they do not, then it is a bug.  I have a question about how the 'Size' column is calculated, will report separate info on that.

Then as an error handling case, we can turn them all off - and the behavior should revert to the default set of datastreams.   

note also that I don't want to over invest as the System Integration package is likely to get a major overhaul soon, you can track updates here, just fyi:  https://github.com/elastic/integrations-dev/issues/152  

FYI - I noticed in my test list at the top that I missed 2 (System Memory and System Process, but since both are on by default we didn't miss any testing).  In comparing, it didn't match the UI when I was reviewing so I was worried - but its all good, all we need to do is update the test steps to include them.

The above represents what I will call now a 'phase 1' of testing.  Phase 2, after we generate the datastreams... we would evaluate the data fields themselves in Discover tab for quality of data ingested under normal circumstances.  And Phase 3 we should make sure the individual options for each dataset are working (if you change the Period time interval or other values) is it respected? 

Can you finish the testing and report back what comes please?  Thanks again!

**Comment by muskangulati-qasource** [comment]
Hi @EricDavisX,

The link you provided https://kibana.endpoint.elastic.dev/app/ingestManager#/integrations/detail/system-0.5.3/overview is not accessible to us.

**Refer Screenshot**:
![error](https://user-images.githubusercontent.com/60252716/93340372-5d9c1580-f84a-11ea-9cd0-2c24b8ab9640.PNG)

**Build details** :
```
Elastic Cloud Staging : 7.10.0
Elastic-Agent URL : https://artifacts-api.elastic.co/v1/search/7.10.0-SNAPSHOT
- Windows : elastic-agent-7.10.0-SNAPSHOT-windows-x86_64.zip
- Linux : elastic-agent-7.10.0-SNAPSHOT-linux-x86_64.tar.gz
- Mac : elastic-agent-7.10.0-SNAPSHOT-darwin-x86_64.tar.gz
```
## Steps :
1. Downloaded the Elastic-agent for respective OS.
2. Deployed the agent on 7.10 cloud build.
3. Added the System Integration in the policy applied on the endpoint.
4. Edited the integration and enabled all the data streams under 'Collect metrics from System instances' tab.
5. Checked for the system integrations showing up for each OS from the Data Stream tab.

## Observations:
## 1. Windows :
**Screenshot** :
![Windows](https://user-images.githubusercontent.com/60252716/93340405-668ce700-f84a-11ea-8983-87f8758238f2.PNG)

**Metrics** :
- [ ] System process metrics
- [ ] System memory metrics
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics

## 2. Linux :
**Screenshot** :
![Linux](https://user-images.githubusercontent.com/60252716/93340421-6ab90480-f84a-11ea-8445-7899d49be6b2.PNG)

**Metrics** :
- [ ] System process metrics
- [ ] System memory metrics
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics

## 3. Mac :
**Screenshot** :
![Mac](https://user-images.githubusercontent.com/60252716/93340429-6c82c800-f84a-11ea-9a3e-93f572cc8723.PNG)

**Metrics** :
- [ ] System process metrics
- [ ] System memory metrics
- [ ] System core metrics
- [ ] System cpu metrics
- [ ] System diskio metrics
- [ ] System entropy metrics
- [ ] System filesystem metrics
- [ ] System fsstat metrics
- [ ] System load metrics
- [ ] System network metrics
- [ ] System network summary metrics
- [ ] System process_summary metrics
- [ ] System raid metrics
- [ ] System service metrics
- [ ] System socket metrics
- [ ] System socket_summary metrics
- [ ] System uptime metrics
- [ ] System users metrics

**Executed 03 Testcases under Testplan**  : System Integration package data streams

Also @EricDavisX ,We are unable to get all the data streams for Windows & Mac. Is it expected? 
If not, can you please provide information on how can we get the same.

Thanks!

**Comment by EricDavisX** [comment]
Hi - It is expected, but the exact ones intended to be supported are listed in the product itself, on the page I posted as an example.  you can log into your Kibana and browse to this URL:
/app/ingestManager#/integrations/detail
 and go to System Integration and look at the long list of text describing which Integrations are supported by which OSes.  Once you walk down the list, you can compare against what you got above and we'll know if anything isn't showing up as expected.

I will follow up with the team about getting your Okta group access to the internal team kibana server

Thanks in advance!

**Comment by muskangulati-qasource** [comment]
Hi @EricDavisX,

Thank you for the information and details, it was really helpful.

Build details :
```
Elastic Cloud Staging : 7.10.0
Elastic-Agent URL : https://artifacts-api.elastic.co/v1/search/7.10.0-SNAPSHOT
- Windows : elastic-agent-7.10.0-SNAPSHOT-windows-x86_64.zip
- Linux : elastic-agent-7.10.0-SNAPSHOT-linux-x86_64.tar.gz
- Mac : elastic-agent-7.10.0-SNAPSHOT-darwin-x86_64.tar.gz
```
### Observations:
### Here is a list of data streams supported by different OSes :
| # | Data Streams  | Supported on Windows | Supported on Mac  | Supported on Linux | Expected Behaviour |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| 1 | Core | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 2 | CPU | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 3 | Disk IO | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 4 | Entropy | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 5 | Filesystem | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 6 | Fsstat | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 7 | Load | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 8 | Memory | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 9 | Network | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 10 | Network_summary | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 11 | Process | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 12 | Process summary | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 13 | RAID | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 14 | Service | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 15 | Socket | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 16 | Socket summary | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 17 | Uptime | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 18 | Users | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |

### Screenshots showing size differences over time [ These are captured every 10 minutes ] :
1. **Windows**
![WindowsSnapshots](https://user-images.githubusercontent.com/60252716/93473020-19c11300-f913-11ea-9a8c-adc59f9e5392.PNG)

2. **Linux**
![LinuxSnapshots](https://user-images.githubusercontent.com/60252716/93473062-2ba2b600-f913-11ea-83bc-1ea9d855727d.PNG)

3. **Mac**
![MacSnapshots](https://user-images.githubusercontent.com/60252716/93473085-34938780-f913-11ea-9bdd-6ddc7dd96f0a.PNG)

We observed that size decreased and increased randomly for all the data streams with respect to time.

### Changing the period for a data stream :
We changed the period for the data streams and observed that it worked as the functionality says.
The data did not updated for that particular data streams until it crossed the period time provided.

`Query : Eric please confirm if it is enough or we need to change any other parameter for testing.`

**Disabled all the data streams** :
We disabled all the data streams under the 'Collect metrics from System instances' tab and then observed changes in size for all the data streams.
1. **Windows** : 
![WindowsoffFinal](https://user-images.githubusercontent.com/60252716/93473234-6573bc80-f913-11ea-969e-d8d96536b02d.png)

2. **Linux**
![LinuxoffFinal](https://user-images.githubusercontent.com/60252716/93473238-673d8000-f913-11ea-9a7a-84d5c6a320b2.PNG)

3. **Mac**
![MacoffFinal](https://user-images.githubusercontent.com/60252716/93473240-69074380-f913-11ea-8070-bfa29602e9d7.png)

We observed that size stopped adding up. However, we the data streams still showed up in the UI under data streams tab.
Note that we used just 1 ES node for 1 Kibana instance(If it matters).

`Query : Eric please confirm if it is expected for data streams disabled but still showing up in UI`

### Checked for all the data streams in the discover tab :
| # | Data Streams  | Windows | Mac  | Linux | Expected Behaviour |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| 1 | event.dataset : "system.core" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 2 | event.dataset : "system.cpu" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 3 | event.dataset : "system.diskio" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 4 | event.dataset : "system.entropy" | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 5 | event.dataset : "system.filesystem" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 6 | event.dataset : "system.fsstat" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 7 | event.dataset : "system.load" | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 8 | event.dataset : "system.memory" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 9 | event.dataset : "system.network" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 10 | event.dataset : "system.network_summary" | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 11 | event.dataset : "system.process" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 12 | event.dataset : "system.process_summary" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 13 | event.dataset : "system.raid" | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 14 | event.dataset : "system.service" | <ul><li>[ ] </li></ul>  | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 15 | event.dataset : "system.socket" | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 16 | event.dataset : "system.socket_summary" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 17 | event.dataset : "system.uptime" | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | <ul><li>[x] </li></ul> | Yes |
| 18 | event.dataset : "system.users" | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> | <ul><li>[x] </li></ul> | Yes |

**Refer screenshot for confirmation** :
![WindowsSystem process](https://user-images.githubusercontent.com/60252716/93473454-ab308500-f913-11ea-9479-548f635a2ba0.png)

**Created 02  Testcases and Executed 06 Testcases under testplan** : [System Integration package data streams](https://elastic.testrail.io/index.php?/runs/view/663)

Please let us know if we are missing anything from our side.

Thanks!

**Comment by EricDavisX** [comment]
here is my reply for the open queries:
Query : Eric please confirm if it is expected for data streams disabled but still showing up in UI
 - yes, this is expected, it is a historical view of all data, so it will not go away when Agent is unenrolled or datastream is disabled, it is reflective of the data in ES

reply about the 'NOTE': we used just one node for ES...
 - it is likely that the duplicate / redundant shards on the same node are still impacting the data variance (getting smaller and larger over time), I need to research if this can be changed via cloud UI or a start up option so we can more easily and directly validate this, at least once.

regarding only changing the time Period value:
Query : Eric please confirm if it is enough or we need to change any other parameter for testing.
 - each parameter that can be tested reasonably should be.  A few are quite involved, so start with the clear and easy ones and then you can report back which ones are NOT covered and we'll discuss that hopefully smaller list.  thanks @muskangulati-qasource

**Comment by muskangulati-qasource** [comment]
Hi @EricDavisX ,

Thank you for looking into our queries and answering the same.

Also, we had a issue that was marked as won't fix stating that for all the fields of the integration, we only test one scenario i.e. the required fields can't be empty. They need to have some default value.
Please check [comment](https://github.com/elastic/kibana/issues/73636#issuecomment-668280422)

We have tested for only required fields, and for optional fields there is no validation.

`Query : There is no validation for the optional fields. Is it expected? Since we are not sure if fields will take input as integer or string. Also, adding any data in those fields, do not result in any change.`

**Build details** :
```
Elastic Cloud Staging : 7.10.0
Elastic-Agent URL : https://artifacts-api.elastic.co/v1/search/7.10.0-SNAPSHOT
- Windows : elastic-agent-7.10.0-SNAPSHOT-windows-x86_64.zip
- Linux : elastic-agent-7.10.0-SNAPSHOT-linux-x86_64.tar.gz
- Mac : elastic-agent-7.10.0-SNAPSHOT-darwin-x86_64.tar.gz
```
### Observations:
### 1. Windows
1. System core metrics
- [] Period
- [] "percentages"
- [] "ticks"

2. System cpu metrics
- [] Period
- [] "normalized_percentages"
- [] "percentages"
- [] "ticks"

3. System diskio metrics
- [] Period
- [] Include Devices [Optional]

4. System filesystem metrics
- [] Period

5. System fsstat metrics
- [] Period

6. System memory metrics
- [] Period

7. System network metrics
- [] Period
- [] Interfaces [Optional]

8. System process metrics
- [] Period
- [] Process Include Top N By Cpu
- [] Process Include Top N By Memory
- [] Enable cmdline cache [Optional]
- [] Enable cgroup reporting [Optional]
- [] Env whitelist [Optional]
- [] Include CPU Ticks [Optional]
- [] Processes

9. System process_summary metrics
- [] Period

10. System socket_summary metrics
- [] Period

11. System uptime metrics
- [] Period

### 2. MacOS
1. System core metrics
- [] Period
- [] "percentages"
- [] "ticks"

2. System cpu metrics
- [] Period
- [] "normalized_percentages"
- [] "percentages"
- [] "ticks"

3. System diskio metrics
- [] Period
- [] Include Devices [Optional]

4. System filesystem metrics
- [] Period

5. System fsstat metrics
- [] Period

6. System load metrics
- [] Period

7. System memory metrics
- [] Period

8. System network metrics
- [] Period
- [] Interfaces [Optional]

9. System process metrics
- [] Period
- [] Process Include Top N By Cpu
- [] Process Include Top N By Memory
- [] Enable cmdline cache [Optional]
- [] Enable cgroup reporting [Optional]
- [] Env whitelist [Optional]
- [] Include CPU Ticks [Optional]
- [] Processes

10. System process_summary metrics
- [] Period

11. System socket_summary metrics
- [] Period

12. System uptime metrics
- [] Period

### 3. Linux
1. System core metrics
- [] Period
- [] "percentages"
- [] "ticks"

2. System cpu metrics
- [] Period
- [] "normalized_percentages"
- [] "percentages"
- [] "ticks"

3. System diskio metrics
- [] Period
- [] Include Devices [Optional]

4. System entropy metrics
- [] Period

5. System filesystem metrics
- [] Period

6. System fsstat metrics
- [] Period

7. System load metrics
- [] Period

8. System memory metrics
- [] Period

9. System network metrics
- [] Period
- [] Interfaces [Optional]

10. System network summary metrics
- [] Period

11. System process metrics
- [] Period
- [] Process Include Top N By Cpu
- [] Process Include Top N By Memory
- [] Enable cmdline cache [Optional]
- [] Enable cgroup reporting [Optional]
- [] Env whitelist [Optional]
- [] Include CPU Ticks [Optional]
- [] Processes

12. System process_summary metrics
- [] Period

13. System raid metrics
- [] Period
- [] Raid Mount Point [Optional]

14. System service metrics
- [] Period
- [] State Filter [Optional]
- [] Pattern Filter [Optional]

15. System socket metrics
- [] Period
- [] Enable Reverse Lookup [Optional]
- [] Reverse Lookup Success TTL [Optional]
- [] Reverse Lookup Success TTL [Optional]

16. System socket_summary metrics
- [] Period

17. System uptime metrics
- [] Period

18. System users metrics
- [] Period

We changed and updated the fields, and found that data was streaming and adding up with respect to time in the size field.

Executed 01 Testcase under testplan : [System Integration package data streams](https://elastic.testrail.io/index.php?/runs/view/663)

Please let us know if we are missing any thing from our end.

Thanks!!

**Comment by EricDavisX** [comment]
@muskangulati-qasource I appreciate the continued dive into this. :)  I accept that the product doesn't enforce any validation on optional fields, but I do think we can assess and test deeper still.

@fearful-symmetry Alex, hi.  I'd like your expertise and thoughts.  We can identify where the regression QA group will cite boundary for its testing, for one thing please feel free to cite that any of this is already covered in automation field by field and we can halt discussion, else we can push on unless there is other reason.  If we are to test any deeper we could use some more info - with regards to testing System Integration, we're able to test all of the fields with default values.  For optional fields, we were not sure in most cases of what valid input values might be, and how to see the alterations in the given Docs put into ES.  Muskan, I think the size increase of the datastream won't be enough here - we'll need to figure out which type of document to look for and how to 'trigger' them if needed.

Alex, we reviewed these links :
https://www.elastic.co/guide/en/beats/metricbeat/current/metricbeat-module-system.html
https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-module-system.html

Do you think we could / should update these docs with more info, or just wait for any coming Integration side docs.  And I can't remember if we had a ticket for porting over the File/Metricbeat module docs to Integration docs, etc.  @ph @ruflin feel free to jump in and we can agree on how deep and how to test.  

Also, I'm going to log a new ticket for the e2e-testing automation, and will likely move this out out of this repo.  :)
